src/foam/i18n/LocaleTranslationService.js

```diff
@@ -20,8 +29,61 @@ foam.CLASS({
     {
       name: 'getTranslation',
       javaCode: `
-      System.err.println("*********************************** getTranslation");
-      return "foobar";
+        if ( locale == "" || source == "" ) return defaultText;
+
+        DAO localeDAO = (DAO) getX().get("localeDAO");
+
+        // localeVariant - complete culture code ('en-CA', ‘fr-CA’, ‘de_AT’ …etc) 
+        // locale - determines locale type (en, fr, es …etc)        
+        // variant - Locale variation (CA for en-CA, CA for fr-CA, AT for de_AT …etc)
+        String variant = "";
+        String localeVariant = "";
```

Comment:
Don't use localeVariant since we are going to remove it. Just use locale and variant.